.github/workflows/go.yml

```diff
@@ -0,0 +1,45 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.18
+
+    - name: Install Packages
+      run: ./install-packages
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+        go get -u golang.org/x/lint/golint
+
+    - name: Build
+      run: make build
+
+    - name: Docker Build
+      run: docker build .
+
+    - name: Test
+      run: make test
+
+    - name: Lint
```

Comment:
can we run lint first? usually it fails faster than tests which take longer to run.
actually, would be great to run this as a commit hook if you know how to set that up already

Reply:
very good point